sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -223,7 +225,11 @@ abstract class JdbcDialect extends Serializable with Logging{
       case f: GeneralAggregateFunc if f.name() == "AVG" =>
         assert(f.inputs().length == 1)
         val distinct = if (f.isDistinct) "DISTINCT " else ""
-        Some(s"AVG($distinct${f.inputs().head})")
+        if (supportCompletePushDown) {
+          Some(s"AVG($distinct${f.inputs().head})")
+        } else {
+          Some(s"Sum($distinct${f.inputs().head}), Count($distinct${f.inputs().head})")
```

Comment:
why do we need it? It's Spark's responsibility to rewrite avg to sum/count, not the data source.

Reply:
If Spark cannot complete pushdown AVG to database, we still want partial pushdown.